src/Illuminate/Database/Connection.php

```diff
@@ -1619,4 +1622,77 @@ public static function getResolver($driver)
     {
         return static::$resolvers[$driver] ?? null;
     }
+
+    /**
+     * Escapes a value for safe SQL embedding.
+     *
+     * @param  string|float|int|bool  $value
+     * @param  bool  $binary
+     * @return string
+     */
+    public function escape($value, $binary = false)
+    {
+        if ($binary) {
+            return $this->escapeBinary($value);
+        } elseif (is_numeric($value)) {
```

Comment:
Hmm, you're right. But adding an enum to again control the mode of escaping makes using it more complex. That `'3e5'` will be used as a string is wrong. You're right. I will fix that. 

I believe not using an enum and escaping the real type without any conversions is more intelligent. Because otherwise you get problems when someone provides `'3d5'` and wants it escaped as a `float`. This is the same type madness PHP had before strict types. I believe doing absolutely no type conversions and therefore no specification of types is the best approach.